Guus/fbl api

guusw opened the pull request:
Changes for formabble C API stuff

<!--
Before submitting your PR, please review the following checklist:

- [ ] **DO NOT** submit a PR without having discussed the change first in a related issue. If none exist, create one first.
- [ ] **CONSIDER** adding a unit test if your PR resolves an issue.
- [ ] **DO** keep pull requests small so they can be easily reviewed.
- [ ] **DO** make sure all unit tests pass.
- [ ] **DO** make sure not to introduce any new compiler warnings.
- [ ] **AVOID** breaking the continuous integration build.
- [ ] **AVOID** making significant changes to the overall architecture.
-->

guusw committed:
Expose shards log init API

guusw committed:
Improve sleep resolution on windows

sinkingsugar commented on shards/core/runtime.hpp:
```diff
@@ -437,7 +466,7 @@ inline void sleep(std::chrono::nanoseconds duration) {
       (void)0;
 #endif
-  } else if (0.0) {
+  } else {
```
the idea here is that if you pass negative you SKIP sleeping totally tho

guusw replied:
I added back that logic for the sleep(double) version to keep it compatible

sinkingsugar reviewed (changes requested):
Some comment btw

sinkingsugar commented on include/shards/shards.h:
```diff
@@ -1323,12 +1347,20 @@ typedef struct _SHCore {
 typedef SHCore *(__cdecl *SHShardsInterface)(uint32_t abi_version);
 
+// Version of the shards interface that does not initialize itself automatically
+typedef SHCore *(__cdecl *SHShardsInterface1)(uint32_t abi_version);
```
why we need this anyway?

guusw replied:
I added this so we can setup the log file before initialization. Since normally ShardsInterface does everything when you call it

sinkingsugar replied:
call it `shardsInterfaceNoInit` maybe?

sinkingsugar replied:
Maybe just breaking change is just fine tbh

guusw replied:
Yeah lemme fix that

sinkingsugar commented on include/shards/shards.h:
```diff
@@ -1126,6 +1128,28 @@ typedef struct SHStringWithLen(__cdecl *SHFastStringLoad)(uint64_t id);
+typedef void(__cdecl *SHLogCallback)(SHStringWithLen category, SHStringWithLen message, int level, void *userData);
+
+// set logToFile to log to the default log file
+#define SHLOG_LOG_TO_DEFAULT_FILE ((const char *)size_t(~0))
```
I agree with the AI

guusw committed:
Add shards init to test runtime

guusw committed:
Add missing calls to shards init

guusw committed:
Add JPH_NO_FORCE_INLINE for emscripten

JPH has a very large inline function which will generate too many locals sometimes inside the wasm

guusw committed:
Fix stderr log initialization on emscripten
